java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/aggregation/AggFuncWithoutRetract.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 The Feathub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf.aggregation;
+
+/**
+ * Parent class of {@link AggFunc} child classes that does not support retractions.
+ *
+ * <p>Some aggregation functions, like {@link
+ * com.alibaba.feathub.flink.udf.aggregation.minmax.MinMaxAggFunc}, have optimized implementations
+ * when retraction is disabled, and this class provides a unified parent class for these
+ * implementation classes.
+ */
+public abstract class AggFuncWithoutRetract<IN_T, OUT_T, ACC_T>
```

Comment:
Would it be simpler to remove this class and let the concrete class implement `throw new UnsupportedOperationException()`?